### DISCUSSION TITLE
Possible to use Tailwind with existing stylesheets?

### DISCUSSION THREAD

**Discussion by fnick851** [tailwindlabs/tailwindcss#5063] [Help]
I am building a SPA that will be embedded into an existing site. The site is already using Bootstrap and several other stylesheets they wrote. I wonder if it is possible to use Tailwind and avoid conflicting with existing styles. 

e.g. 
1. An option for Tailwind CSS to somehow be "scoped"? 
2. Maybe I just need to somehow remove the base resets (preflight)?

Has anyone tried to use Tailwind in such a scenario?

**Comment by dance2die** [comment]
I was messing with Docusaurus and it was conflicting with infima, the [default Docusaurus style library](https://docusaurus.io/docs/styling-layout#styling-your-site-with-infima).

So what I did was to "scope" by [prefixing](https://tailwindcss.com/docs/configuration#prefix) with `tw-` and removed the "base" (preflight) as you wrote (because Docusaurus depended on browser default paddings/margins).

It worked but abandoned as light/dark theme changes are too hardwired w/ infima.  
At least the approach worked.

**Reply by simonswiss** [reply]
Yep, that's pretty accurate. Your options to make Tailwind work better with other existing libraries are:

- add a `prefix` that will be applied to every utility to avoid collisions 
- make adjustments to the `@base` layer
- use the `important` option in the config (or, if using the `JIT` mode, use the `!` prefix on a per-class basis) if you're having specificity issues

Also, make sure you're loading Tailwind CSS _last_, so you have less risks of having your utility classes overridden by Bootstrap and your own custom styles.
